tests/conftest.py

```diff
@@ -860,3 +861,17 @@ def factory(dirpath: pathlib.Path, read_bytes=True) -> dict:
         return serialized
 
     return factory
+
+
+@pytest.fixture
```

Comment:
This adds quite a bit of (unnecessary) overhead. I think that in some cases where `/usr/bin/bash` is hardcoded, the actual path doesn't actually matter. It is not actually validated to be an existing path. So for all those cases, you are now unnecessarily calling a subprocess.

A first improvement would be to make it a session scoped fixture so it is executed only once per session. I think that should work. But I would prefer to only use this fixture for tests where the path really needs to exist. For all other tests where it is just some path, I think it is better to simply hard code it.

Reply:
```suggestion
@pytest.fixture(scope="session")
```

Reply:
> But I would prefer to only use this fixture for tests where the path really needs to exist. For all other tests where it is just some path, I think it is better to simply hard code it.

For what reason? The computational cost should be the same once it is a session fixture since it is only created once?

Reply:
If it is a session fixture it is ok, I agree